modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/webpack_federation.js

```diff
@@ -12,61 +12,198 @@
  * details.
  */
 
-Liferay = window.Liferay || {};
+(function() {
+	const CONTAINER_REQUESTS_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_CONTAINER_REQUESTS__');
+	const CONTAINERS_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_CONTAINERS__');
+	const GET_MODULE_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_GET_MODULE__');
+	const SHARED_SCOPE_SYMBOL = Symbol.for('__LIFERAY_WEBPACK_SHARED_SCOPE__');
 
-Liferay.Webpack = {
-	Container: {},
-	SharedScope: {},
-};
+	window[CONTAINER_REQUESTS_SYMBOL] = window[CONTAINER_REQUESTS_SYMBOL] || {};
+	window[CONTAINERS_SYMBOL] = window[CONTAINERS_SYMBOL] || {};
+	window[SHARED_SCOPE_SYMBOL] = window[SHARED_SCOPE_SYMBOL] || {};
 
-Liferay.require = (moduleName) =>
-	new Promise((resolve, reject) => {
-		const i = moduleName.indexOf('/');
+	const containerRequests = window[CONTAINER_REQUESTS_SYMBOL];
+	const sharedScope = window[SHARED_SCOPE_SYMBOL];
 
-		let containerId, path;
+	/**
+	 * Create a new container request
+	 */
+	function createContainerRequest(url, onLoadHandler) {
+		const script = document.createElement('script');
 
-		if (i === -1) {
-			containerId = moduleName;
-			path = '.';
-		}
-		else {
-			containerId = moduleName.substring(0, i);
-			path = moduleName.substring(i + 1);
-		}
+		const containerRequest = {
+			// set if request failed
+			error: undefined,
+			// set to true if the .js file has been fetched from the server
+			fetched: false,
+			// set to the exported value (if error is set module is undefined)
+			module: undefined,
+			// set to the <script> DOM node while it is being retrieved (undefined after)
+			script,
+			// set to all people that needs to be notified once the <script> loads
```

Comment:
needs → need

But I don't think you mean "people" here anyway...

Reply:
:sweat_smile: